File: Leetcode/401. Binary Watch/binary_watch.py
#
# @lc app=leetcode id=401 lang=python3
#
# [401] Binary Watch
#

# @lc code=start
class Solution:
    def readBinaryWatch(self, turnedOn: int):
        res=[]
        for h in range(12):
            for m in range(60):
                if(bin(h).count("1")+bin(m).count("1")==turnedOn):
                    res.append(f"{h}:{m:02d}")
        return res
        
# @lc code=end
sol = Solution()
print(sol.readBinaryWatch(2))

File: Leetcode/401. Binary Watch/binary_watch.rs
struct Solution{}

impl Solution {
    pub fn read_binary_watch(turned_on: i32) -> Vec<String> {
        let mut result = Vec::new();
        let nu = turned_on as u32;
        for h in 0i32..12{
            for m in 0i32..60{
                if (h.count_ones() + m.count_ones()) == nu {
                    result.push(format!("{}:{:02}", h, m));
                }
            }
        }
        result
    }
}
fn main() {
    let res = Solution::read_binary_watch(2);
    println!("{:?}",res);
}
